### PR TITLE
feat: Form.Item support `initialValue` & `getValueProps`

### LINKED ISSUE #19727: Antd 4.0, Form.Item 使用 normalize 和 getValueFromEvent 转换控件的赋值取值方式, onFinish 返回的值依旧是转换前的值
- [ ] I have searched the [issues](https://github.com/ant-design/ant-design/issues) of this repository and believe that this is not a duplicate.

### Reproduction link
[Edit on CodeSandbox](https://codesandbox.io/s/antd-y7klb)

### Steps to reproduce
1.选择日期
2.提交表单
3.观察控制台日志

### What is expected?
以DatePicker为例, 我想要让表单字段以字符串的形式对组件进行控制, 而不是moment对象, 目的是减少表单数据初始化及数据提交时需要额外对数据类型的转换;
1.onFinish事件, 返回的date字段值的是getValueFromEvent返回的值
2.initialValues中的date期望可以先通过normalize转换字段值再给控件

### What is actually happening?
1.onFinish提交时, 返回值依旧是moment对象;
2.initialValues以字符串的格式给DatePicker的表单字段初始化时报类型错误

| Environment | Info |
|---|---|
| antd | 4.0.0-alpha.8 |
| React | 16.11.0 |
| System | macOS 10.14.2 |
| Browser | Chrom 78 |

### LINKED ISSUE #21816: antd v4 add initialValue support on Form.Item like v3
- [ ] I have searched the [issues](https://github.com/ant-design/ant-design/issues) of this repository and believe that this is not a duplicate.

### What problem does this feature solve?
解决复杂表单，循环渲染表单列表时赋值 

### What does the proposed API look like?
目前from 是将默认值挂载到form 上 ，期望还是和v3一样挂载在每个具体的item上

-----

@zombieJ  Update:
# Hi all. This issue is welcomed for feedback collection. If you like add `initialValue` on Form.Item, pls add 👍 on the topic directly. Or 👎 if not agree. Add feel free for other suggestion.

### LINKED ISSUE #22896: A way to decouple form values from input values in Ant 4 (getValueProps and getValueFromEvent in ant 3)
- [x] I have searched the [issues](https://github.com/ant-design/ant-design/issues) of this repository and believe that this is not a duplicate.

### What problem does this feature solve?
Imagine I have a datetime field.  
```
<Form.Item
    label="Encerra em"
    name="expires_at"
>
    <DatePicker
        showTime={{ format: 'HH:mm' }}
        format="DD/MM/YYYY HH:mm"
    />
</Form.Item>
```

The input value is a Moment object.  
I can not send a moment object via a request, so I have to create a function to change the "expires_at" value before sending in the form values.  

In ant 3, I could use getValueFromEvent to transform the moment to a string and getValueProps to transform the string to a moment object.  
That way, the field's value (moment object) was "translated" to the form's value (a string).  

In ant 4 there's no "getValueProps" in Form.Item, so there's no way to do that.

### What does the proposed API look like?
```
<Form.Item
    // The name does not need to be like this
    transformFormValue={form_value => input_value}
>
```

### PR DESCRIPTION
**Pull request by zombieJ**
### 🤔 This is a ...

- [x] New feature
- [ ] Bug fix
- [ ] Site / document update
- [ ] Component style update
- [ ] TypeScript definition update
- [ ] Refactoring
- [ ] Code style optimization
- [ ] Test Case
- [ ] Branch merge
- [ ] Other (about what?)

### 🔗 Related issue link

resolve #21816
resolve #22896
resolve #19727

### 📝 Changelog

| Language   | Changelog |
| ---------- | --------- |
| 🇺🇸 English |   Form.Item support `initialValue` and `getValueProps` props.        |
| 🇨🇳 Chinese |    Form.Item 支持 `initialValue` 和 `getValueProps` 属性。       |

### ☑️ Self Check before Merge

⚠️ Please check all items below before review. ⚠️ 

- [x] Doc is updated/provided or not needed
- [x] Demo is updated/provided or not needed
- [x] TypeScript definition is updated/provided or not needed
- [x] Changelog is provided or not needed

-----
[View rendered components/form/index.en-US.md](https://github.com/ant-design/ant-design/blob/field-enhancement/components/form/index.en-US.md)
[View rendered components/form/index.zh-CN.md](https://github.com/ant-design/ant-design/blob/field-enhancement/components/form/index.zh-CN.md)

### CODE REVIEW

File: components/form/index.zh-CN.md [hunk 1]
```diff
@@ -68,30 +68,32 @@ const validateMessages = {
...
+| normalize | 转换字段值给控件 | (value, prevValue, prevValues) => any | - |  |
+| required | 是否必填，如不设置，则会根据校验规则自动生成 | boolean | false |  |
+| rules | 校验规则，设置字段的校验逻辑。点击[此处](#components-form-demo-basic)查看示例 | [Rule](#Rule)[] | - |  |
+| shouldUpdate | 自定义字段更新逻辑，说明[见下](#shouldUpdate) | boolean \| (prevValue, curValue) => boolean | false |  |
+| trigger | 设置收集字段值变更的时机 | string | onChange |  |
+| validateFirst | 当某一规则校验不通过时，是否停止剩下的规则的校验 | boolean | false |  |
+| validateStatus | 校验状态，如不设置，则会根据校验规则自动生成，可选：'success' 'warning' 'error' 'validating' | string | - |  |
+| validateTrigger | 设置字段校验的时机 | string \| string[] | onChange |  |
+| valuePropName | 子节点的值的属性，如 Switch 的是 'checked'。该属性为 `getValueProps` 的封装，自定义 `getValueProps` 后会失效 | string | 'value' |  |
+| wrapperCol | 需要为输入控件设置布局样式时，使用该属性，用法同 `labelCol`。你可以通过 Form 的 `wrapperCol` 进行统一设置。当和 Form 同时设置时，以 Item 为准。 | [object](/components/grid/#Col) | - |  |
```
Comment (afc163) [inline components/form/index.zh-CN.md:96]: 感觉这个月会有很多冲突。

Comment (afc163) [reply components/form/index.zh-CN.md:96]: 月底前来加版本吧，或者给 master 先加上一列再 rebase 进来。

Comment (zombieJ) [reply components/form/index.zh-CN.md:96]: 👌

Comment (zombieJ) [reply components/form/index.zh-CN.md:96]: done

**Review by afc163 — APPROVED** [review]

**Comment by crazyair** [comment]
image

**Comment by yoyo837** [comment]
@crazyair 没发布修复。

**Pull request merged**